subsys/usb/device_next/usbd_core.c

```diff
@@ -183,23 +193,28 @@ static ALWAYS_INLINE void usbd_event_handler(struct usbd_context *const uds_ctx,
 	}
 }
 
-static void usbd_thread(void *p1, void *p2, void *p3)
+static void usbd_flush_queue(void)
 {
-	ARG_UNUSED(p1);
-	ARG_UNUSED(p2);
-	ARG_UNUSED(p3);
-
+	static struct k_mutex mutex;
 	struct usbd_context *uds_ctx;
 	struct udc_event event;
+	int ret;
+
+	k_mutex_lock(&mutex, K_FOREVER);
```

Comment:
This will result in an ASSERT.

Reply:
I must have avoided this by chance by only testing on one particular hardware.